### ISSUE TITLE
CollidableBox Bug with AnimationPlayer

### ISSUE THREAD

**Issue by spikeystar** [spikeystar/GGRPG#5]
I am trying to make an item box that moves and animates when touched. However, the current CollidableBox rejects an AnimationPlayer child node- any time I click away from the scene or close the editor it deletes the AnimationPlayer, even if I had saved it.

<img width="1511" alt="Screenshot 2024-01-21 at 4 46 43 PM" src="https://github.com/spikeystar/GGRPG/assets/145728899/3791620a-ed52-41fc-9b0d-bb2e841b8038">

**Comment by Giwayume** [comment]
Yeah, the script manages a bunch of children that it creates on the fly, that's probably why it deletes any other nodes added. Shouldn't be too hard to fix.

**Comment by spikeystar** [comment]
I was able to get around the issue of not being able to add child nodes by making the CollidableBox and others all child nodes of a Node2D as a new scene (PresentBase.tscn)

However, the collisions do not appear to work properly when animated- I set the height very high to demonstrate what it is doing. I have also tried animating the height value to follow the box, but this does not seem to help.

Below is a screenshot where I also tried implementing this with a moving platform and have the same issue.

<img width="581" alt="Screenshot 2024-04-24 at 11 10 57 AM" src="https://github.com/spikeystar/GGRPG/assets/145728899/5367b513-9faa-49a2-b2e3-48f264c06942">
